### ISSUE TITLE
Indentation: false positive for one-line lambda

### ISSUE THREAD

**Issue by pbludov** [checkstyle/checkstyle#5675]
`/var/tmp $ javac TestClass.java`:

`/var/tmp $ cat TestClass.java`:

```
import java.util.stream.Stream;
class TestClass {
    void method() {
        Stream.of(false).map(x -> { if (x) return 0; else return 1; }); // Line 6
    }
}
```

`/var/tmp $ cat config.xml`:

```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="Indentation"/>
  </module>
</module>
```

`/var/tmp $ java -Duser.language=en -Duser.country=US -jar checkstyle-8.8-all.jar -c config.xml TestClass.java`:

```
Starting audit...
[ERROR] /var/tmp/TestClass.java:6: 'if' has incorrect indentation level 36, expected level should be 12. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```

---------------

There should be no violations. 

--------------

**Comment by pbludov** [comment]
Origin:
http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/test/java/util/SplittableRandom/SplittableRandomTest.java#l451

**Comment by pbludov** [comment]
This is not related to the lambda. The shortest example is:
```
class TestClass {
    void method(boolean x) {
        { if (x) return; }  //  'if' has incorrect indentation level 10, expected level should be 12. [Indentation]
    }
}
```

**Comment by pbludov** [comment]
The problematic code is here:
```
    /**
     * Check the indent of the top level token.
     */
    protected void checkTopLevelToken() {
        final DetailAST topLevel = getTopLevelAst();

        if (topLevel != null
                && !getIndent().isAcceptable(expandedTabsColumnNo(topLevel))
                && !hasLabelBefore()
                && (shouldTopLevelStartLine() || isOnStartOfLine(topLevel))) {
            logError(topLevel, "", expandedTabsColumnNo(topLevel));
        }
    }
    /**
     * Determines if the top level token must start the line.
     *
     * @return true
     */
    protected boolean shouldTopLevelStartLine() {
        return true;
    }

```

The indentation check simply wants a line wrap here. This is not what the indentation check should take care of. The git says that this code was here from the first commit. I think that this is a rudiment from those ancient times, when there was  no `LeftCurly` check. I vote to remove the outdated code.
